icu4c/source/common/uloc.cpp

```diff
@@ -914,6 +914,7 @@ uloc_setKeywordValue(const char* keywordName,
     startSearchHere = (char*)locale_getKeywordsStart(buffer);
     if(startSearchHere == NULL || (startSearchHere[1]==0)) {
         if(keywordValueLen == 0) { /* no keywords = nothing to remove */
+            U_ASSERT(*status != U_STRING_NOT_TERMINATED_WARNING);
```

Comment:
The assertions in this function just before it returns are potentially harmful because you didn't check at function entry that the caller didn't give you a not-terminated warning. This is public API, so could be called from anyone, and a previous ICU function call may have left that warning. (That's one of the problems with our warnings, and why we stopped using them in newer API / for newer purposes.)

Please remove these new assertions from this function.